### DISCUSSION TITLE
Is there a reason for not implementing IDisposable?

### DISCUSSION THREAD

**Discussion by mungk** [adoconnection/RazorEngineCore#75] [General]
Hi, I've been looking over the code and trying to determine if there's a reason that RazorEngineCompiledTemplate doesn't implement IDisposable given that it owns a MemoryStream. I can't find a reason so I'm curious if it's intentional and I'm missing something or if it's something that should be added. Thanks!

**Comment by wdcossey** [comment]
From Microsoft:

'''
This type (MemoryStream) implements the IDisposable interface, but does not actually have any resources to dispose. This means that disposing it by directly calling Dispose() or by using a language construct such as using (in C#) or Using (in Visual Basic) is not necessary.
'''

**Reply by mungk** [reply]
Interesting, I've not noticed that before. Thanks for pointing it out. Really curious would they would do this though. Given their own guidance that you should dispose disposables, why make extra work for everyone (like me) who didn't spot that note in the docs?

**Reply by wdcossey** [reply]
I was recently reading up on MemoryStream, particularly around disposing it, realised that it didn't matter if I did/didn't, clearing the 'byte[]' was a better solution (in my scenario).

If you would like some more info, below is a good StackOverflow article.

https://stackoverflow.com/a/4274769
